src/Orleans.TestingHost/LegacyTestClusterConfiguration.cs

```diff
@@ -82,20 +82,28 @@ public static ClientConfiguration BuildClientConfiguration(ClusterConfiguration
 
         internal void ConfigureLegacyConfiguration(IConfigurationBuilder configBuilder)
         {
-            SerializationManager serializationManager = CreateLegacyConfigurationSerializer();
-
-            var configSource = new Dictionary<string, string>();
-            if (this.builder.Properties.TryGetValue(ClusterConfigurationKey, out object clusterConfig))
+            // Add configuration overrides to the cluster configuration because some code relies on this information.
+            // TODO: rewrite code which relies on these overrides and remove this block.
             {
-                configSource[ClusterConfigurationKey] = Serialize(serializationManager, clusterConfig);
-            }
+                var configuration = configBuilder.Build();
+                var options = new TestClusterOptions();
+                configuration.Bind(options);
 
-            if (this.builder.Properties.TryGetValue(ClientConfigurationKey, out object clientConfig))
-            {
-                configSource[ClientConfigurationKey] = Serialize(serializationManager, clientConfig);
+                for (var i = 0; i < this.builder.InitialSilosCount; i++)
+                {
+                    var siloName = options.GetSiloSpecificOptions(i).SiloName;
+
+                    this.ClusterConfiguration.GetOrCreateNodeConfigurationForSilo(siloName);
```

Comment:
Why go back to defining these? can't you avoid it and only do so specifically if a particular test was incorrectly relying on it?
I'd like to prevent re-adding this at the infrastructure level if it was just for an edge case test

Reply:
I was hoping to avoid it, too, but it's used in non-test code in a non-trivial way which was making tests fail. I had a fix for the one test usage I sent to you, but I reverted it and went with this approach.

https://github.com/dotnet/orleans/blob/9bd486bb7368a83a37b7565e2d267aed959e4df7/src/Orleans.Runtime/Streams/QueueBalancer/StaticClusterDeploymentConfiguration.cs#L26

It will [have to] be addressed when we move away from ClusterConfiguration internally